lib/Saml2/Utils.php

```diff
@@ -348,6 +389,26 @@ public static function setSelfHost($host)
     }
 
     /**
+     * @param $baseurlpath string The baseurl path to use when constructing URLs
+     */
+    public static function setBaseURLPath($baseurlpath)
+    {
+        if (!isset($baseurlpath) || empty($baseurlpath)) {
+            $baseurlpath = '/';
+        }
+
+        self::$_baseurlpath = '/'. ltrim(rtrim($baseurlpath, '/') . '/', '/');
```

Comment:
I wanted to assure that all start and end with a /. What the reason for avoiding that?